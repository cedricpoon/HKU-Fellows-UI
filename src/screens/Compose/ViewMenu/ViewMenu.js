import React, { Component } from 'react';
import { Icon, Text } from 'native-base';
import PropTypes from 'prop-types';

import { PopupMenu, PopupMenuItem } from 'hkufui/components';
import NavigationService from 'hkufui/src/NavigationService';
import { localize } from 'hkufui/locale';
import styles from 'hkufui/theme/Styles';

const locale = localize({ language: 'en', country: 'hk' });

class ViewMenu extends Component {
  constructor(props) {
    super(props);
  }

  _showPreview = (anonymity) => {
    this._popup.toggle();
    NavigationService.navigate('ComposePreview', {
      anonymity,
      ...this.props.payload
    });
  }

  render() {
    const { onRef, ...restProps } = this.props;

    return(
      <PopupMenu
       /* Proptypes handling on <PopupMenu /> */
        ref={ref => { this._popup = ref; onRef(ref) }}
        { ...restProps }
      >
        <PopupMenuItem transparent success iconLeft onPress={() => { this._showPreview(false) }}>
          <Icon name="face" type="MaterialIcons" style={styles.icon}></Icon>
          <Text>{locale['new.nonAnonymous']}</Text>
        </PopupMenuItem>
        <PopupMenuItem transparent info iconLeft onPress={() => { this._showPreview(true) }}>
          <Icon name="incognito" type="MaterialCommunityIcons" style={styles.icon}></Icon>
          <Text>{locale['new.anonymous']}</Text>
        </PopupMenuItem>
      </PopupMenu>
    );
  }
}

ViewMenu.propTypes = {
  onRef: PropTypes.func,
  payload: PropTypes.shape({
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
    hashtags: PropTypes.object,
    content: PropTypes.string.isRequired
  }).isRequired
}

export default ViewMenu;
