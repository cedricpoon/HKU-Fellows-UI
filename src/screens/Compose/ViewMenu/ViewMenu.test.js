import React from 'react';
import { shallow } from 'enzyme';
import ViewMenu from './ViewMenu';

describe('Testing ViewMenu Screen', () => {
  const requiredProps = {
    position: {x: 0, y: 0},
    toggle: () => {},
    payload: { title: 'title', content: 'content' }
  };

  it('renders as expected', () => {
    const wrapper = shallow(
      <ViewMenu {...requiredProps} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
