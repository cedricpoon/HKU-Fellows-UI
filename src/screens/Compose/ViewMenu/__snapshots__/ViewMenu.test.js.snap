// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ViewMenu Screen renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewMenu
    position={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    toggle={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Button)
          iconLeft={true}
          success={true}
          transparent={true}
        >
          <Styled(Icon)
            name="face"
            style={
              Object {
                "textAlign": "center",
                "width": 25,
              }
            }
            type="MaterialIcons"
          />
          <Styled(Text)>
            Post with UID
          </Styled(Text)>
        </Styled(Button)>,
        <Styled(Button)
          iconLeft={true}
          info={true}
          transparent={true}
        >
          <Styled(Icon)
            name="incognito"
            style={
              Object {
                "textAlign": "center",
                "width": 25,
              }
            }
            type="MaterialCommunityIcons"
          />
          <Styled(Text)>
            Post Anonymously
          </Styled(Text)>
        </Styled(Button)>,
      ],
      "position": Object {
        "x": 0,
        "y": 0,
      },
      "toggle": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Icon)
              name="face"
              style={
                Object {
                  "textAlign": "center",
                  "width": 25,
                }
              }
              type="MaterialIcons"
            />,
            <Styled(Text)>
              Post with UID
            </Styled(Text)>,
          ],
          "iconLeft": true,
          "success": true,
          "transparent": true,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "face",
              "style": Object {
                "textAlign": "center",
                "width": 25,
              },
              "type": "MaterialIcons",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Post with UID",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "Post with UID",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Icon)
              name="incognito"
              style={
                Object {
                  "textAlign": "center",
                  "width": 25,
                }
              }
              type="MaterialCommunityIcons"
            />,
            <Styled(Text)>
              Post Anonymously
            </Styled(Text)>,
          ],
          "iconLeft": true,
          "info": true,
          "transparent": true,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "incognito",
              "style": Object {
                "textAlign": "center",
                "width": 25,
              },
              "type": "MaterialCommunityIcons",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Post Anonymously",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "Post Anonymously",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Button)
            iconLeft={true}
            success={true}
            transparent={true}
          >
            <Styled(Icon)
              name="face"
              style={
                Object {
                  "textAlign": "center",
                  "width": 25,
                }
              }
              type="MaterialIcons"
            />
            <Styled(Text)>
              Post with UID
            </Styled(Text)>
          </Styled(Button)>,
          <Styled(Button)
            iconLeft={true}
            info={true}
            transparent={true}
          >
            <Styled(Icon)
              name="incognito"
              style={
                Object {
                  "textAlign": "center",
                  "width": 25,
                }
              }
              type="MaterialCommunityIcons"
            />
            <Styled(Text)>
              Post Anonymously
            </Styled(Text)>
          </Styled(Button)>,
        ],
        "position": Object {
          "x": 0,
          "y": 0,
        },
        "toggle": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Icon)
                name="face"
                style={
                  Object {
                    "textAlign": "center",
                    "width": 25,
                  }
                }
                type="MaterialIcons"
              />,
              <Styled(Text)>
                Post with UID
              </Styled(Text)>,
            ],
            "iconLeft": true,
            "success": true,
            "transparent": true,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "face",
                "style": Object {
                  "textAlign": "center",
                  "width": 25,
                },
                "type": "MaterialIcons",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Post with UID",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": "Post with UID",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Icon)
                name="incognito"
                style={
                  Object {
                    "textAlign": "center",
                    "width": 25,
                  }
                }
                type="MaterialCommunityIcons"
              />,
              <Styled(Text)>
                Post Anonymously
              </Styled(Text)>,
            ],
            "iconLeft": true,
            "info": true,
            "transparent": true,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "incognito",
                "style": Object {
                  "textAlign": "center",
                  "width": 25,
                },
                "type": "MaterialCommunityIcons",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Post Anonymously",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": "Post Anonymously",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
